src/components/tasks/card/TaskStatusEditMode.tsx

```diff
@@ -49,38 +67,42 @@ const TaskStatusEditMode = ({
         if (newProgress !== undefined) {
             payload.percentCompleted = newProgress;
         }
+
+        const taskStatusUpdatePromise =
+            isDevMode && isSelfTask
+                ? updateSelfTask({ id: task.id, task: payload })
+                : updateTask({ id: task.id, task: payload });
+
         setEditedTaskDetails((prev: CardTaskDetails) => ({
             ...prev,
             ...payload,
         }));
-        const response = updateTask({
-            id: task.id,
-            task: payload,
-        });
 
-        response
-            .unwrap()
-            .then(() => {
-                setSaveStatus(SAVED);
-            })
-            .catch(() => {
-                setSaveStatus(ERROR_STATUS);
-            })
-            .finally(() => {
-                setTimeout(() => {
-                    setSaveStatus('');
-                }, 3000);
-            });
+        try {
+            await taskStatusUpdatePromise.unwrap();
+            setSaveStatus(SAVED);
+            toast(SUCCESS, TASK_STATUS_UPDATE_SUCCESS_MESSAGE);
+        } catch (error: any) {
+            setSaveStatus(ERROR_STATUS);
+            const errorMessage =
+                error?.data?.message ?? TASK_STATUS_UPDATE_ERROR_MESSAGE;
+            toast(ERROR, errorMessage);
```

Comment:
- there is no need to create a variable here unnecessarily, it can be done inline instead
example: 
```tsx
toast(ERROR, error?.data?.message ?? TASK_STATUS_UPDATE_ERROR_MESSAGE)
```